### ISSUE TITLE
For Tutorial , NameError name 'null' is not defined.

### ISSUE THREAD

**Issue by jiahaozhou** [scrapy/scrapy#3475]
I just start with scrapy, I followed the tutorial, after enter   scrapy crawl quotes    I got an error.

(snowflakes) C:\Users\Jiaha\tutorial>scrapy crawl quotes
Traceback (most recent call last):
  File "C:\Users\Jiaha\AppData\Local\conda\conda\envs\snowflakes\Scripts\scrapy-script.py", line 10, in <module>
    sys.exit(execute())
  File "C:\Users\Jiaha\AppData\Local\conda\conda\envs\snowflakes\lib\site-packages\scrapy\cmdline.py", line 149, in execute
    cmd.crawler_process = CrawlerProcess(settings)
  File "C:\Users\Jiaha\AppData\Local\conda\conda\envs\snowflakes\lib\site-packages\scrapy\crawler.py", line 249, in __init__
    super(CrawlerProcess, self).__init__(settings)
  File "C:\Users\Jiaha\AppData\Local\conda\conda\envs\snowflakes\lib\site-packages\scrapy\crawler.py", line 137, in __init__
    self.spider_loader = _get_spider_loader(settings)
  File "C:\Users\Jiaha\AppData\Local\conda\conda\envs\snowflakes\lib\site-packages\scrapy\crawler.py", line 336, in _get_spider_loader
    return loader_cls.from_settings(settings.frozencopy())
  File "C:\Users\Jiaha\AppData\Local\conda\conda\envs\snowflakes\lib\site-packages\scrapy\spiderloader.py", line 61, in from_settings
    return cls(settings)
  File "C:\Users\Jiaha\AppData\Local\conda\conda\envs\snowflakes\lib\site-packages\scrapy\spiderloader.py", line 25, in __init__
    self._load_all_spiders()
  File "C:\Users\Jiaha\AppData\Local\conda\conda\envs\snowflakes\lib\site-packages\scrapy\spiderloader.py", line 47, in _load_all_spiders
    for module in walk_modules(name):
  File "C:\Users\Jiaha\AppData\Local\conda\conda\envs\snowflakes\lib\site-packages\scrapy\utils\misc.py", line 71, in walk_modules
    submod = import_module(fullpath)
  File "C:\Users\Jiaha\AppData\Local\conda\conda\envs\snowflakes\lib\importlib\__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "C:\Users\Jiaha\tutorial\tutorial\spiders\quotes_spider.py", line 33, in <module>
    "execution_count": null,
NameError: name 'null' is not defined

(snowflakes) C:\Users\Jiaha\tutorial>scrapy crawl quotes
Traceback (most recent call last):
  File "C:\Users\Jiaha\AppData\Local\conda\conda\envs\snowflakes\Scripts\scrapy-script.py", line 10, in <module>
    sys.exit(execute())
  File "C:\Users\Jiaha\AppData\Local\conda\conda\envs\snowflakes\lib\site-packages\scrapy\cmdline.py", line 149, in execute
    cmd.crawler_process = CrawlerProcess(settings)
  File "C:\Users\Jiaha\AppData\Local\conda\conda\envs\snowflakes\lib\site-packages\scrapy\crawler.py", line 249, in __init__
    super(CrawlerProcess, self).__init__(settings)
  File "C:\Users\Jiaha\AppData\Local\conda\conda\envs\snowflakes\lib\site-packages\scrapy\crawler.py", line 137, in __init__
    self.spider_loader = _get_spider_loader(settings)
  File "C:\Users\Jiaha\AppData\Local\conda\conda\envs\snowflakes\lib\site-packages\scrapy\crawler.py", line 336, in _get_spider_loader
    return loader_cls.from_settings(settings.frozencopy())
  File "C:\Users\Jiaha\AppData\Local\conda\conda\envs\snowflakes\lib\site-packages\scrapy\spiderloader.py", line 61, in from_settings
    return cls(settings)
  File "C:\Users\Jiaha\AppData\Local\conda\conda\envs\snowflakes\lib\site-packages\scrapy\spiderloader.py", line 25, in __init__
    self._load_all_spiders()
  File "C:\Users\Jiaha\AppData\Local\conda\conda\envs\snowflakes\lib\site-packages\scrapy\spiderloader.py", line 47, in _load_all_spiders
    for module in walk_modules(name):
  File "C:\Users\Jiaha\AppData\Local\conda\conda\envs\snowflakes\lib\site-packages\scrapy\utils\misc.py", line 71, in walk_modules
    submod = import_module(fullpath)
  File "C:\Users\Jiaha\AppData\Local\conda\conda\envs\snowflakes\lib\importlib\__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "C:\Users\Jiaha\tutorial\tutorial\spiders\quotes_spider.py", line 33, in <module>
    "execution_count": null,
NameError: name 'null' is not defined

Code:      # I exactly followed the tutorial.
import scrapy
class QuotesSpider(scrapy.Spider):
    name = "quotes"
    
    def start_requests(self):
        urls = [
            'http://quotes.toscrape.com/page/1/',
            'http://quotes.toscrape.com/page/2/',
        ]
        for url in urls:
            yield scrapy.Request(url=url,callback=self.parse)
            
    def parse(self,response):
        page = response.url.split("/")[-2]
        filename = 'quotes-%s.html' % page
        with open(filename, 'wb') as f:
            f.write(response.body)
        self.log('Saved file %s' % filename)

**Comment by pel-h** [comment]
Hello @jiahaozhou ,

I have checked this and it seems ok. I also checked from a jupyter notebook. See attached.  
[3475.zip](https://github.com/scrapy/scrapy/files/2589403/3475.zip)

Looks like more a problem with your environment or something like that, to get support on this post it in StackOverflow as the topics here are for issues or change request.

In all cases double check what you did and good luck.

**Comment by wRAR** [comment]
Hello @jiahaozhou , the traceback says you have an error in the line 33 yet the file you provided doesn't have that line, so I suspect you provided the wrong file and not `C:\Users\Jiaha\tutorial\tutorial\spiders\quotes_spider.py`.

**Comment by datasharq** [comment]
This is an old post, but I just ran into the same problem today with Python 3.7 / Jupyter Lab. In my scenario I had an original Notebook with several cells containing model classes, and it was getting hard to read. I saved several cells as a new Notebook and tried to import that. Fail. Then I renamed the Notebook into a Python file and tried to import THAT. That's when I got the same error as the OP.

Solution: Restarted the kernel of the Notebook containing the `import` statements. That's it. Until you restart the Kernel, it doesn't look for the new files. The `null` error is coming from iPython.
